notebooks/oracle2bq/OracleToBigQuery_notebook.ipynb

```diff
@@ -0,0 +1,888 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "06735281-1c22-48d3-9b71-20f9ca99d04a",
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "! python ./setup.py bdist_egg --output=$PACKAGE_EGG_FILE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9e1a779f-2c39-42ec-98be-0f5e9d715447",
+   "metadata": {},
+   "source": [
+    "#### Copying JARS files to GCS_STAGING_LOCATION"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "939cdcd5-0f3e-4f51-aa78-93d1976cb0f4",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "! gsutil cp main.py $GCS_STAGING_LOCATION/dependencies/\n",
+    "! gsutil cp -r $PACKAGE_EGG_FILE $GCS_STAGING_LOCATION/dependencies/\n",
+    "! gsutil cp spark-bigquery-with-dependencies_2.12-0.27.0.jar $GCS_STAGING_LOCATION/jars/spark-bigquery-with-dependencies_2.12-0.27.0.jar\n",
+    "! gsutil cp ojdbc8-21.7.0.0.jar $GCS_STAGING_LOCATION/jars/ojdbc8-21.7.0.0.jar"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d9bb170-09c4-40d1-baaf-9e907f215889",
+   "metadata": {},
+   "source": [
+    "## Step 8: Calculate Parallel Jobs for ORACLE to BIGQUERY\n",
+    "This step uses MAX_PARALLELISM parameter to calculate number of parallel jobs to run"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2c501db0-c1fb-4a05-88b8-a7e546e2b1d0",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# calculate parallel jobs:\n",
+    "COMPLETE_LIST = copy.deepcopy(ORACLETABLE_LIST)\n",
+    "PARALLEL_JOBS = len(ORACLETABLE_LIST)//MAX_PARALLELISM\n",
+    "JOB_LIST = []\n",
+    "while len(COMPLETE_LIST) > 0:\n",
+    "    SUB_LIST = []\n",
+    "    for i in range(MAX_PARALLELISM):\n",
+    "        if len(COMPLETE_LIST)>0 :\n",
+    "            SUB_LIST.append(COMPLETE_LIST[0])\n",
+    "            COMPLETE_LIST.pop(0)\n",
+    "        else:\n",
+    "            break\n",
+    "    JOB_LIST.append(SUB_LIST)\n",
+    "print(\"list of tables for execution : \")\n",
+    "print(JOB_LIST)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "78f6f83b-891a-4515-a1d6-f3406a25dc2a",
+   "metadata": {},
+   "source": [
+    "## Step 9: Execute Pipeline to Migrate tables from ORACLE to BIGQUERY\n",
+    "- BIGQUERY_DATASET : Target dataset in Bigquery\n",
+    "- BIGQUERY_MODE : Mode of operation at target append/overwrite\n",
+    "- TEMP_GCS_BUCKET : Bucket name for dataproc job staging\n",
+    "- PYTHON_FILE_URIS : Path to PACKAGE_EGG_FILE\n",
+    "- MAIN_PYTHON_CLASS : Path to main.py"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1f98914b-bd74-4d0e-9562-7019d504a25e",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "oracle_to_gcs_jobs = []\n",
+    "BIGQUERY_DATASET=\"oracle_dump\"\n",
+    "BIGQUERY_MODE = \"append\"\n",
```

Comment:
default mode should be overwrite.
Also specify other write modes commented, next to it